### ISSUE TITLE
Unable to resolve module ./factory.js

### ISSUE THREAD

**Issue by okevino47** [eemeli/intl-pluralrules#34]
Hello, i have a problem, i am installing i18next in a react native project.
I am trying to use i18next with plurals feature but that does not work actually

**_What did i do :_**

By doing so i am specifying in the i18next initialization `compatibilityJSON: 'v4',` (like below) :

```
const initI18next = async () => {
  await i18n
    .use(initReactI18next)
    // init i18next
    // for all options read: https://www.i18next.com/overview/configuration-options
    .init({
      resources,
      fallbackLng: 'en',
      debug: true,
      compatibilityJSON: 'v4',
      interpolation: {
        escapeValue: false,
      },
    });
};
```

When i build my application, a warning message appear -> `i18next::pluralResolver: Your environment seems not to be Intl API compatible, use an Intl.PluralRules polyfill. Will fallback to the compatibilityJSON v3 format handling`

According to some stack overflow found and documentation :
- https://stackoverflow.com/questions/70493788/i18nextpluralresolver-your-environment-seems-not-to-be-intl-api-compatible-u
- https://github.com/eemeli/intl-pluralrules
- https://www.i18next.com/misc/migration-guide

Finally after installing myself plural-rules

I have this error : 

`
error: Error: Unable to resolve module ./factory from [PROJECT_PATH]/mobile/node_modules/intl-pluralrules/plural-rules.js: ./factory could not be found within the project or in these directories:
  node_modules

> 3 | var getPluralRules = require('./factory');

`

 and i don't find any solution on the web actually

**Comment by eemeli** [comment]
Try updating your Node.js, it might be a bit old?

**Comment by eemeli** [comment]
Closing as presumably answered.

**Comment by codaDevDan** [comment]
I am having the same issue actually. Using node 16.13.0 so I don't think that is the cause? It is strange because when I look for the file './factory.js', it is there, and has an export. Any help would be much appreciated. Came here from i18next for context: 
https://www.i18next.com/misc/json-format#i-18-next-json-v3

**Comment by eemeli** [comment]
@codaDevDan Can you provide step-by-step instructions for a reproduction of the issue?

**Comment by codaDevDan** [comment]
Yes:

1. create an expo app with ```expo init```
2. downgrade to expo 46 by modifying the package.json expo version (our app is running 46)
3. make sure dependencies are in line with expo 46 by running ```npx expo install --check```
4. install your library
5. import the library in any file (App.tsx is what I tried) ```import "intl-pluralrules";```
6. npx expo start. The app will crash and you will see the following error message:

```
iOS Bundling failed 46ms
Unable to resolve "./factory.js" from "node_modules/intl-pluralrules/plural-rules.js"
```

**Comment by eemeli** [comment]
@codaDevDan I'm not an `expo` user, so I ran the following commands:

```sh
npx create-expo-app # expo init produces error, recommends this instead
cd my-app # the default name
npm install expo@46
npm install intl-pluralrules
echo 'import "intl-pluralrules";' >> App.js
npx expo start
```

Then pointing a web browser at `http://localhost:19000/node_modules/expo/AppEntry.bundle?platform=ios&dev=true&hot=false` resulted in a JS file that appears to properly contain the `intl-pluralrules` module, and the CLI output includes:

```
iOS Bundling complete 6190ms
```

At this point you might need to provide something like a Dockerfile which reproduces your error, as it may be something specific to your environment or other configuration.

**Comment by codaDevDan** [comment]
Yeah, that is possible.  Upgrading the expo SDK to 48 (latest version) seems solves the issue, so it could also be a module resolution problem of some kind. Unfortunately I don't have time to do what you suggested here, so I am going to try to upgrade. Thanks for the quick response!

**Comment by flygomel** [comment]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `intl-pluralrules@2.0.0` for the project I'm working on.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/intl-pluralrules/plural-rules.js b/node_modules/intl-pluralrules/plural-rules.js
index 7a54d00..56b6c2a 100644
--- a/node_modules/intl-pluralrules/plural-rules.js
+++ b/node_modules/intl-pluralrules/plural-rules.js
@@ -1,6 +1,6 @@
 'use strict';
 
-var getPluralRules = require('./factory.js');
+var getPluralRules = require('./factory');
 
 function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }
 
```

**Comment by skdrew** [comment]
Hi, I've just upgraded our Expo app to SDK 49 today and this exact issue has affected us as well.

I came to the same conclusion as @flygomel to patch the problem, but after digging into it a little more the problem may actually be in the exports from package.json.

I've just changed this section:

```
  "react-native": {
    "./factory": "./factory.js"
  },
```

to

```
  "react-native": {
    "./factory.js": "./factory.js"
  },
```

which has fixed the module resolution without changing any code.

I'm not certain what the correct way of handling this is to suit both modules, but more than happy to help out any suggestions to fix it for everyone.
